main.tf

```diff
@@ -134,6 +135,8 @@ locals {
 
       runners_instance_ready_command = var.runner_worker_docker_autoscaler.instance_ready_command
 
+      use_private_key = var.runner_worker.use_private_key && var.runner_worker.type == "docker-autoscaler"
```

Comment:
thought: Sounds weired that we use `var.runner_worker.type == "docker-autoscaler"` here as this is the config for the autoscaler. But the same statement can be found above.